packages/frontend/core/src/modules/cloud/services/websocket.ts

```diff
@@ -14,24 +16,40 @@ export class WebSocketService extends Service {
   });
   sockets: Set<Socket> = new Set();
 
-  constructor() {
+  constructor(private readonly authService: AuthService) {
     super();
   }
 
-  newSocket(): Socket {
+  /**
+   * Create a new socket, with automatic connect and reconnect logic.
+   * External code should not call `socket.connect()` or `socket.disconnect()` manually.
+   * When socket is no longer needed, call `dispose()` to clean up resources.
+   */
+  createSocket() {
     const socket = this.ioManager.socket('/');
     this.sockets.add(socket);
+    if (this.authService.session.status$.value === 'authenticated') {
+      socket.connect();
+    }
 
-    return socket;
+    return {
+      socket,
+      dispose: () => {
+        socket.disconnect();
+        this.sockets.delete(socket);
```

Comment:
ioManager will reuse sockets, so there will always be only one socket instance

Reply:
They just share the same underlying websocket connection, but have independent event listeners

Reply:
![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/5c0ymolP9B7QStCsS1RP/59c29891-8b37-4da2-ba43-663d189f3344.png)

always the same instance